include/deal.II/lac/petsc_vector_base.h

```diff
@@ -347,17 +347,31 @@ namespace PETScWrappers
      *
      * To figure out which elements exactly are stored locally, use
      * local_range().
+     *
+     * @deprecated use locally_owned_size() instead.
      */
+    DEAL_II_DEPRECATED
     size_type
     local_size() const;
 
+    /**
+     * Return the local dimension of the vector, i.e. the number of elements
+     * stored on the present MPI process. For sequential vectors, this number
+     * is the same as size(), but for parallel vectors it may be smaller.
+     *
+     * To figure out which elements exactly are stored locally, use
+     * local_range().
```

Comment:
That's not wrong, but I think it's a limitation to the PETSc vectors that they only store contiguous ranges and that what we should recommend is to query `locally_owned_elements()` instead. Maybe list both?